pkg/apps/controller/deployer/deployer_controller.go

```diff
@@ -262,9 +262,16 @@ func (c *DeploymentController) handle(deployment *corev1.ReplicationController,
 		}
 
 	case appsv1.DeploymentStatusComplete:
-		if err := c.cleanupDeployerPods(deployment); err != nil {
+		// preserve the deployer pod for the last successful deployment
+		ok, err := c.isLastCompletedDeployment(deployment)
+		if err != nil {
 			return err
 		}
+		if !ok {
```

Comment:
Where?  cleanupDeployerPods affects a single RC.  This will never delete the pod from a completed deployment unless there is a newer completed deployment.